src/direct-binary-upload-process.js

```diff
@@ -12,238 +12,159 @@ governing permissions and limitations under the License.
+
+        await this.partUploader.uploadParts(uploadResult, initResponse.getAllParts(), this);
+
+        uploadResult.stopTimer();
+
+        // output json result to logger
+        this.logInfo('Uploading result in JSON: ' + JSON.stringify(uploadResult, null, 4));
+    }
+
+    /**
+     * Does the work of initiating an upload by calling the initiateUpload servlet.
+     *
+     * @param {UploadResult} uploadResult Result to which information about the upload will be added.
+     * @param {Array} [uploadFiles] If specified, will be used in place of the upload files in the
+     *  process's upload options.
+     * @returns {InitResponse} Information about the response from the initiateUpload servlet.
      */
-    async upload() {
+    async initiateUpload(uploadResult, uploadFiles = false) {
         let options = this.getUploadOptions();
         const url = options.getUrl();
         const targetFolder = options.getTargetFolderPath();
-        const toUpload = options.getUploadFiles().map(upload => new UploadFile(this.getOptions(), options, upload));
-        const headers = options.getHeaders();
-        const concurrent = options.isConcurrent();
+        const targetFiles = uploadFiles ? uploadFiles : options.getUploadFiles();
+        const toUpload = targetFiles.map(upload => new UploadFile(this.getOptions(), options, upload));
 
-        const uploadResult = new UploadResult(this.getOptions(), options);
         uploadResult.startTimer();
 
-        const initOptions = {
-            url: `${url}.initiateUpload.json`,
-            method: 'POST',
-            headers: {
-                ...headers,
-                'content-type': 'application/x-www-form-urlencoded',
-            },
-            data: querystring.stringify({
+        const initRequest = new HttpRequest(this.getOptions(), `${url}.initiateUpload.json`)
+            .withMethod(HttpRequest.Method.POST)
+            .withContentType('application/x-www-form-urlencoded')
+            .withData(querystring.stringify({
                 path: targetFolder,
                 fileName: toUpload.map(file => file.getFileName()),
                 fileSize: toUpload.map(file => file.getFileSize()),
-            }),
-            responseType: 'json',
-        };
+            }))
+            .withResponseType(HttpRequest.ResponseType.JSON)
+            .withUploadOptions(this.getUploadOptions());
 
         let initResponse;
         let fileListInit;
 
-        try {
-            this.initCancelToken = createCancelToken();
-            const response = await timedRequest(initOptions, this.getRetryOptions(options, uploadResult), this.initCancelToken);
-            this.initCancelToken = null;
-            const {
-                data: resObj,
-                status: statusCode,
-                elapsedTime = 0,
-            } = response;
-            options = updateOptionsWithResponse(options, response);
+        this.logInfo(`Initiating upload to ${url}`);
```

Comment:
I would make this a debug log - logging presigned url is frowned upon in prod (we got it as feedback in Nui security review - we shouldn't log presigned urls).

Reply:
I double-checked that the url is a non-signed URL (so we should be ok here). I went ahead and changed the level to `debug`, though, just to be safe.